--- x265-2.7/build/linux/multilib.sh	2018-02-21 03:55:56.000000000 -0500
+++ src/build/linux/multilib.sh	2018-06-26 12:57:34.000000000 -0400
@@ -4,17 +4,17 @@
 
 cd 12bit
 cmake ../../../source -DHIGH_BIT_DEPTH=ON -DEXPORT_C_API=OFF -DENABLE_SHARED=OFF -DENABLE_CLI=OFF -DMAIN12=ON
-make ${MAKEFLAGS}
+make 
 
 cd ../10bit
 cmake ../../../source -DHIGH_BIT_DEPTH=ON -DEXPORT_C_API=OFF -DENABLE_SHARED=OFF -DENABLE_CLI=OFF
-make ${MAKEFLAGS}
+make 
 
 cd ../8bit
 ln -sf ../10bit/libx265.a libx265_main10.a
 ln -sf ../12bit/libx265.a libx265_main12.a
-cmake ../../../source -DEXTRA_LIB="x265_main10.a;x265_main12.a" -DEXTRA_LINK_FLAGS=-L. -DLINKED_10BIT=ON -DLINKED_12BIT=ON
-make ${MAKEFLAGS}
+cmake ../../../source -DEXTRA_LIB="x265_main10.a;x265_main12.a" -DEXTRA_LINK_FLAGS=-L. -DLINKED_10BIT=ON -DLINKED_12BIT=ON -DCMAKE_INSTALL_PREFIX=../../../../install -DENABLE_SHARED:bool=off
+make 
 
 # rename the 8bit library, then combine all three into libx265.a
 mv libx265.a libx265_main.a
@@ -36,6 +36,7 @@
 else
 
 # Mac/BSD libtool
-libtool -static -o libx265.a libx265_main.a libx265_main10.a libx265_main12.a 2>/dev/null
-
+ /usr/bin/libtool -static -o libx265.a libx265_main.a libx265_main10.a libx265_main12.a 2>/dev/null
 fi
+
+make install
